task.sql

```diff
@@ -1,33 +1,40 @@
--- Create database and tables
-
 CREATE DATABASE ShopDB;
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
-    Name VARCHAR(50),
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50) NOT NULL,
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE WhAddress (
+    ID INT AUTO_INCREMENT,
+    WarehouseName VARCHAR(50) NOT NULL,
+    WarehouseAddress VARCHAR(50) NOT NULL,
     PRIMARY KEY (ID)
 );
 
 CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
-    WarehouseAmount INT,
-    WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
+    ID INT AUTO_INCREMENT,
+    ProductName VARCHAR(50) NOT NULL,
+    WarehouseAmount INT NOT NULL,
+    WarehouseID INT,
+    FOREIGN KEY (WarehouseID) REFERENCES WhAddress(ID) ON DELETE NO ACTION,
     CountryID INT,
 	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
     PRIMARY KEY (ID)
 );
 
--- Populate test data
+INSERT INTO WhAddress (WarehouseName, WarehouseAddress)
+VALUES ('Warehouse-1', 'City-1, Street-1'),
+       ('Warehouse-2', 'City-2, Street-2');
+
+-- Вставка данных в Countries
+INSERT INTO Countries (Name)
+VALUES ('Country1'),
+       ('Country2');
 
-INSERT INTO Countries (ID,Name)
-	VALUES (1, 'Country1');
-INSERT INTO Countries (ID,Name)
-	VALUES (2, 'Country2');
-    
-INSERT INTO ProductInventory (ID,ProductName,WarehouseAmount,WarehouseName,WarehouseAddress,CountryID)
-	VALUES (1, 'AwersomeProduct', 2, 'Warehouse-1', 'City-1, Street-1',1);
-INSERT INTO ProductInventory (ID,ProductName,WarehouseAmount,WarehouseName,WarehouseAddress,CountryID)
-	VALUES (2, 'AwersomeProduct', 5, 'Warehouse-2', 'City-2, Street-2',2);
+-- Вставка данных в ProductInventory
+INSERT INTO ProductInventory (ProductName, WarehouseAmount, WarehouseID, CountryID)
+VALUES ('AwersomeProduct', 2, 1, 1),
+       ('AwersomeProduct', 5, 2, 2);
```

Comment:
There is a typo in the product name 'AwersomeProduct'. It should be corrected to 'AwesomeProduct'.